docs/modules/ROOT/pages/references/architecture/sli_reporting.adoc

```diff
@@ -0,0 +1,341 @@
+
+====== `GET /downtime/`
+
+Lists all available downtime windows.
+
+Query parameters:
+
+* `from`: ISO timestamp defining the start time from which to include downtime windows.
+* `to`: ISO timestamp defining the end time until which to include downtime windows.
+
+The result contains all downtime windows that overlap with the time interval `[from, to]`.
+Notably, this also includes downtime windows that start before `from` but end after `from` (and conversely for `to`), as well as downtime windows that completely encompass the specified time interval.
+
+Returns:
+
+A list of downtime window records (as defined above).
+
+====== `GET /downtime/cluster/:cluster-id`
+
+Lists all available downtime windows that apply to a specific cluster (according to the match rules in `Affects`).
+
+Path parameters:
+* `cluster-id`: Lieutenant ID of the cluster for which to retrieve downtime windows.
+
+Query parameters:
+
+* `from`: ISO timestamp defining the start time from which to include downtime windows.
+* `to`: ISO timestamp defining the end time until which to include downtime windows.
+
+The result contains all downtime windows that overlap with the time interval `[from, to]`.
+Notably, this also includes downtime windows that start before `from` but end after `from` (and conversely for `to`), as well as downtime windows that completely encompass the specified time interval.
+
+Returns:
+
+A list of downtime window records (as defined above).
+
+====== `POST /downtime`
+
+Creates a new downtime window.
+
+Body parameters:
+
+The body of the request corresponds to the JSON example given above, minus the `ID` property.
+Any parameters marked as optional may be omitted.
+
+If an `ExternalID` is provided in the request body, and an existing downtime window shares the same `ExternalID`, then the existing window should be updated (replaced) instead of a new one created.
+Otherwise, a new record is created from the request body and assigned a new random ID.
+
+Returns:
+
+The newly created record, including ID.
+
+====== `POST /downtime/:id`
+
+Updates (replaces) an existing downtime window.
+
+Path parameters:
+* `id`: ID of the existing downtime window record to update.
+
+Body parameters:
+
+The body of the request corresponds to the JSON example given above.
+Any parameters marked as optional may be omitted.
+
+If the `ExternalID` property is modified, a check must be made to ensure the new `ExternalID` doesn't conflict with any existing record.
+If a conflict is found, return an error 400.
+
+Returns:
+
+The newly updated record, including ID.
+
+====== `PATCH /downtime/:id`
+
+Updates an existing downtime window, supporting partial updates.
+
+Path parameters:
+* `id`: ID of the existing downtime window record to update.
+
+Body parameters:
+
+The body of the request corresponds to the JSON example given above.
+Any parameters marked as optional may be omitted.
```

Comment:
For `PATCH` it should be possible to omit all parameters (so we can just patch the start **or** end of a downtime for example), otherwise it's exactly the same as `POST`.